src/tensor_ops.jl

```diff
@@ -1,3 +1,15 @@
+# Deprecate `*` and `'*` for tensor multiplication
```

Comment:
Not sure what I should have done with these methods...
Since `AbstractTensor{order, dim, T <: Real} <: AbstractArray{T, order}`, `*` will still work so I had to stop it somehow! I suspect there is a more correct way to fix this? :)